lib/block_view_validator.go

```diff
@@ -114,9 +114,23 @@ func (validatorEntry *ValidatorEntry) RawEncodeWithoutMetadata(blockHeight uint6
 		data = append(data, EncodeByteArray(domain)...)
 	}
 
+	// VotingPublicKey
+	var votingPublicKeyBytes []byte
+	if validatorEntry.VotingPublicKey != nil {
+		votingPublicKeyBytes = validatorEntry.VotingPublicKey.ToBytes()
```

Comment:
should we implement the DeSoEncoder interface for BLS stuff? It would simplify some of these encode/decode methods

Reply:
I can't define the DeSoEncoder functions on the `bls.*` structs in lib and IMO it breaks encapsulation (and/or creates a circular dependency) to do it in the `bls/` package. For that reason, I refactored these into `Encode` and `Decode` helpers instead which covers the nil check.